pkg/data_sources/privatelink_endpoint.go

```diff
@@ -0,0 +1,88 @@
+package data_sources
+
+import (
+	"context"
+
+	"github.com/gthesheep/terraform-provider-dbt-cloud/pkg/dbt_cloud"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var privatelinkEndpointSchema = map[string]*schema.Schema{
+	"id": &schema.Schema{
+		Type:        schema.TypeString,
+		Computed:    true,
+		Description: "The internal ID of the PrivateLink Endpoint",
+	},
+	"name": &schema.Schema{
+		Type:        schema.TypeString,
+		Optional:    true,
+		Default:     "",
+		Description: "Given descriptive name for the PrivateLink Endpoint (name and/or private_link_endpoint_url need to be provided to return data for the datasource)",
+	},
+	"type": &schema.Schema{
+		Type:        schema.TypeString,
+		Computed:    true,
+		Description: "Type of the PrivateLink Endpoint",
+	},
+	"private_link_endpoint_url": &schema.Schema{
```

Comment:
Is this not required?

Reply:
Ahh I see it needs name or url? Is there a way to validate one or the other is set via the plugin framework?

Reply:
Yes, it is name or URL, or both.

In the Framework we could use [this](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework-validators/resourcevalidator#AtLeastOneOf) to validate but not on the SDK v2. As you saw, I am doing the check when calling the API to make sure that one of them is set.